droidlet/perception/craftassist/heuristic_perception.py

```diff
@@ -569,17 +579,20 @@ def perceive(self, force=False):
             )
             perceive_info["in_perceive_area"]["holes"] = holes if holes else None
             # 1.3 Get all air-touching blocks in perception area
-            blocktypes, shifted_c, tags = get_nearby_airtouching_blocks(
-                self.agent,
-                pos,
-                self.block_data,
-                self.color_data,
-                self.block_property_data,
-                color_list,
-                radius,
-            )
-            if tags and len(shifted_c) > 0:
-                perceive_info["in_perceive_area"]["airtouching_blocks"] = [shifted_c, tags]
+            if self.mark_airtouching_blocks:
+                blocktypes, shifted_c, tags = get_nearby_airtouching_blocks(
+                    self.agent,
+                    pos,
+                    self.block_data,
+                    self.color_data,
+                    self.block_property_data,
+                    color_list,
+                    radius,
+                )
+                if tags and len(shifted_c) > 0:
+                    perceive_info["in_perceive_area"]["airtouching_blocks"] = list(
+                        zip(shifted_c, tags)
+                    )
```

Comment:
and test for this function just so its doesn't slip in if it breaks again

Reply:
same